crates/torii/core/src/model.rs

```diff
@@ -144,11 +150,154 @@ pub fn parse_sql_model_members(model: &str, model_members_all: &[SqlModelMember]
     parse_sql_model_members_impl(model, model_members_all)
 }
 
+/// Creates a query that fetches all models and their nested data.
+pub fn build_sql_query(model_schemas: &Vec<Ty>) -> Result<String, Error> {
+    fn parse_struct(
+        path: &str,
+        schema: &Struct,
+        selections: &mut Vec<String>,
+        tables: &mut Vec<String>,
+    ) {
+        for child in &schema.children {
+            match &child.ty {
+                Ty::Struct(s) => {
+                    let table_name = format!("{}${}", path, s.name);
+                    parse_struct(&table_name, s, selections, tables);
+
+                    tables.push(table_name);
+                }
+                _ => {
+                    // alias selected columns to avoid conflicts in `JOIN`
+                    selections.push(format!(
+                        "{}.external_{} AS \"{}.{}\"",
+                        path, child.name, path, child.name
+                    ));
+                }
+            }
+        }
+    }
+
+    let primary_table = model_schemas[0].name();
+    let mut global_selections = Vec::new();
+    let mut global_tables = model_schemas
+        .iter()
+        .enumerate()
+        .filter(|(index, _)| *index != 0) // primary_table don't `JOIN` itself
+        .map(|(_, schema)| schema.name())
+        .collect::<Vec<String>>();
+
+    for ty in model_schemas {
+        let schema = ty.as_struct().expect("schema should be struct");
+        let model_table = &schema.name;
+        let mut selections = Vec::new();
+        let mut tables = Vec::new();
+
+        parse_struct(model_table, schema, &mut selections, &mut tables);
+
+        global_selections.push(selections.join(", "));
+        global_tables.extend(tables);
+    }
+
+    // TODO: Fallback to subqueries, SQLite has a max limit of 64 on 'table 'JOIN'
```

Comment:
I think it is fine to return an error in this case. 64 joins is a reasonable limit